internal/characters/sara/sara.go

```diff
@@ -0,0 +1,64 @@
+package sara
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gsim/pkg/character"
+	"github.com/genshinsim/gsim/pkg/core"
+)
+
+func init() {
+	core.RegisterCharFunc("sara", NewChar)
+}
+
+type char struct {
+	*character.Tmpl
+	a4LastProc int
+	c1LastProc int
+}
+
+func NewChar(s *core.Core, p core.CharacterProfile) (core.Character, error) {
+	c := char{}
+	t, err := character.NewTemplateChar(s, p)
+	if err != nil {
+		return nil, err
+	}
+	c.Tmpl = t
+	c.Energy = 80
+	c.EnergyMax = 80
+	c.Weapon.Class = core.WeaponClassBow
+	c.NormalHitNum = 5
+	c.BurstCon = 3
+	c.SkillCon = 5
+
+	if c.Base.Cons == 6 {
+		c.c6()
+	}
+
+	return &c, nil
+}
+
+// Handles Sara c6
+func (c *char) c6() {
+	c.Core.Events.Subscribe(core.OnAttackWillLand, func(args ...interface{}) bool {
+		ds := args[1].(*core.Snapshot)
+		if c.Core.Status.Duration(fmt.Sprintf("sarabuff%v", ds.Actor)) <= 0 {
+			return false
+		}
+		if ds.Element != core.Electro {
```

Comment:
i would do this check first instead of the other one. fmt.Sprintf is a bit more expensive and since this function gets called every single attack on every target can add up